observability/tracing/README.md

```diff
@@ -0,0 +1,103 @@
+
+When this is enabled, the following changes occur to your deployment:
+* A network policy is enabled which allows egress to the Jaeger app services
+* The istio-proxy sidecar is configured to sample 100% of incoming requests
+
+This will have the effect that any incoming requests to your application will generate a trace to Jaeger. These traces can be viewed in tracing.*cluster-name*.nais.io, eg [tracing.dev-gcp.nais.io](https://tracing.dev-gcp.nais.io/).
+
+### Propagating trace headers
+
+To give Jaeger sufficient context to reconstruct a trace, the application must read a set of HTTP headers from the incoming request and forward them to any requests further down the chain.
+
+The following trace headers must be forwarded as they are received:
+
+* `x-b3-traceid`
+* `x-b3-spanid`
+* `x-b3-parentspanid`
+* `x-b3-sampled`
+* `x-b3-flags`
+* `x-b3`
+
+The `traceid` identifies a single trace. `spanid` identifies a span. For all spans below top-level, `parentspanid` identifies the span's parent.
+
+![Illustration of the relationship between trace, span and parentspan IDs](trace-span-ids.png)
+
+*See also [Istio's documentation](https://istio.io/latest/faq/distributed-tracing/#how-to-support-tracing) for more detailed definitions of the headers*
+
+<!-- TODO: Check source code to find which headers are actually dealt with by istio-proxy -->
+
+<!-- TODO: Figure out semantics of the sampled/flags headers --> 
+
+### Code examples
+
+Tracing was initially deployed in NAIS in collaboration with team digisos, who used envoy-based tracing to identify an intermittent problem causing spikes in page render time. 
+The examples below are therefore examples, not references.
+Readers who find more elegant solutions are encouraged to submit pull requests to this documentation.
+
+#### React
+
+The following code generates a 128-byte UUID, which is split into two 64-bit identifiers. These identifiers are then passed along with API requests. This creates a trace which spans an entire page session.
+
+<!-- TODO: Clarify jaeger's requirements for identifier; documentation is unclear/contradictory -->
+
+```
+// In the file top level
+import { v4 as uuidv4 } from 'uuid';
+const sessionTraceID = uuidv4().toString().replaceAll('-','');
+
+        // Inside the request method
+        headers.append('X-B3-TraceId', sessionTraceID.substr(0, 16));
+        headers.append('X-B3-SpanId', sessionTraceID.substr(16, 16));
+```
+
+## Direct submission
+
+The user may send tracing data to jaeger via [a plethora of supported protocols](https://www.jaegertracing.io/docs/1.20/apis/), including OpenTracing, Thrift and Zipkin.
+A [robust ecosystem of libraries](https://www.jaegertracing.io/docs/1.20/client-libraries/) exists for several languages.
+
+This mode of using Jaeger is not currently supported by NAIS, but if there is a need, it would be easy to implement.
+This is largely mentioned here to see if there is any demand.
+If there is, please contact Tore Sinding Bekkedal.
```

Comment:
Maybe add appropriate Slack channel instead of single-point-of-reference? Bus factor, potential future documentation maintenance work, and so on?

You @toresbe could instead pay special attention to the mentioned (nais derivative?) slack-channel? =)

Reply:
Good point. I refer them to #nais